gaspy/atoms_operators.py

```diff
@@ -251,6 +252,172 @@ def find_adsorption_sites(atoms):
+
+
+def _plane_normal(coords):
+    """
+    Return the surface normal vector to a plane of best fit
+    by performing planar regression.
+    See https://gist.github.com/amroamroamro/1db8d69b4b65e8bc66a6
+    for the method.
+
+    Arg:
+        coords   A `numpy.ndarray` (n,3),
+                 coordinates of atoms on the slab surface.
+
+    Output:
+        vector  numpy.ndarray. Adsorption vector for an adsorption site.
+    """
+    A = np.c_[coords[:, 0], coords[:, 1], np.ones(coords.shape[0])]
+    vector, _, _, _ = scipy.linalg.lstsq(A, coords[:, 2])
+    vector[2] = -1.0
+    vector /= -np.linalg.norm(vector)
+    return vector
+
+def _ang_between_vectors(v1, v2):
+  
+    """
+    Returns the angle in degree
+    between 3D vectors 'v1' and 'v2'
+
+    Arg:
+        v1    3D vector in np.array(x1,y1,z1) form,
+              the origin is (0,0,0).
+        v1    3D vector in np.array(x2,y2,z2) form,
+              the origin is (0,0,0).
+
+    Output:
+        angle  angle in degrees.
+    """
+    cosang = np.dot(v1, v2)
+    sinang = np.linalg.norm(np.cross(v1, v2))
+    # np.arctan2(sinang, cosang) is angle in radian
+    radian = np.arctan2(sinang, cosang)
+    angle = radian * 57.2958
+    return angle
+
+
+def find_adsorption_vector(bulk_cn_dict, slab_atoms, surface_indices, adsorption_site):
+    """
+    Returns the vector of an adsorption site representing the
+    furthest distance from the neighboring atoms.
+    The vector is a (1,3) numpy array.
+    The idea comes from CatKit.
+    https://catkit.readthedocs.io/en/latest/?badge=latest
+
+    Arg:
+        bulk_cn_dict         A dictionary of coordination numbers
+                             for each distinct site in the respective bulk structure
+        slab_atoms           The `ase.Atoms` format of a supercell slab.
+        surface_indices      The index of the surface atoms in a list.
+        adsorption_site      A `numpy.ndarray` object that contains the x-y-z coordinates
+                             of the adsorptions sites.
+
+    Output:
+        vector            numpy.ndarray. Adsorption vector for an adsorption site.
+    """
+    vnn = VoronoiNN(allow_pathological=True)
+
+    slab_atoms += Atoms('U', [adsorption_site])
+    U_index = slab_atoms.get_chemical_symbols().index('U')
+    struct_with_U = AseAtomsAdaptor.get_structure(slab_atoms)
+    nn_info = vnn.get_nn_info(struct_with_U, n=U_index)
+    nn_indices = [neighbor['site_index'] for neighbor in nn_info]
+    surface_nn_indices = [idx for idx in nn_indices if idx in surface_indices]
+
+    # get the index of the closest 4 atom to the site to form a plane
+    # chose 4 because it will gaurantee a more accurate plane for edge cases
+    nn_dists_from_U = {idx: np.linalg.norm(slab_atoms[idx].position-slab_atoms[U_index].position) 
+                       for idx in surface_nn_indices}
+    sorted_dists = {k: v for k, v in sorted(nn_dists_from_U .items(), key=lambda item: item[1])}
```

Comment:
Use more explicit variable names. I know `k` is short for `key` and `v` is short for `value`, but what are the keys? What are the values? What are the `items`? I think they keys are indices for atoms and values are distance. But the reader should not have to *try* to figure that out. It should be obvious.